jerry-core/ecma/builtin-objects/ecma-builtin-reflect.c

```diff
@@ -77,18 +86,139 @@ ecma_builtin_reflect_dispatch_routine (uint16_t builtin_routine_id, /**< built-i
   JERRY_UNUSED (this_arg);
   JERRY_UNUSED (arguments_number);
 
+  if (builtin_routine_id < ECMA_REFLECT_OBJECT_CONSTRUCT)
+  {
+    /* 1. */
+    if (arguments_number == 0 || !ecma_is_value_object (arguments_list[0]))
+    {
+      return ecma_raise_type_error (ECMA_ERR_MSG ("Argument is not an Object."));
+    }
+
+    /* 2. */
+    ecma_string_t *name_str_p = ecma_op_to_prop_name (((arguments_number > 1) ? arguments_list[1]
+                                                                              : ECMA_VALUE_UNDEFINED));
+
+    /* 3. */
+    if (name_str_p == NULL)
+    {
+      return ECMA_VALUE_ERROR;
+    }
+
+    ecma_value_t ret_value;
+    ecma_object_t *target_p = ecma_get_object_from_value (arguments_list[0]);
+    switch (builtin_routine_id)
+    {
+      case ECMA_REFLECT_OBJECT_GET:
+      {
+        ecma_value_t receiver = arguments_list[0];
+
+        /* 4. */
+        if (arguments_number > 2)
+        {
+          receiver = arguments_list[2];
+        }
+
+        ret_value = ecma_op_object_get_with_receiver (target_p, name_str_p, receiver);
+        break;
+      }
+
+      case ECMA_REFLECT_OBJECT_HAS:
+      {
+        ret_value = ecma_make_boolean_value (ecma_op_object_has_property (target_p, name_str_p));
+        break;
+      }
+
+      case ECMA_REFLECT_OBJECT_DELETE_PROPERTY:
+      {
+        ret_value = ecma_op_object_delete (target_p, name_str_p, false);
+        break;
+      }
+
+      case ECMA_REFLECT_OBJECT_OWN_KEYS:
+      {
+        ret_value = ecma_builtin_helper_object_get_properties (target_p, ECMA_LIST_NO_OPTS);
```

Comment:
Do we have tests for Symbol property names? IMO `ECMA_LIST_NO_OPTS` should be adjusted to list symbol properties as well.